### PR TITLE
Prohibit duplicate pseudo-headers.

### PR DESCRIPTION
**Pull request by nharper**
This fixes issue #883.

### CODE REVIEW

**Review by Lukasa — COMMENTED** [review]
This seems like a reasonable addition, but I think it’s in the wrong place, for two reasons:

1. There’s no particular reason to exclude response pseudo-headers from this prohibition: multiple `:status` fields is just as meaningless.
2. We probably don’t want to prohibit this _generally_, but instead only for the pseudo-headers we define in this document.

To that end I think I recommend moving this up to the parent section, and noting something like “All pseudo-header fields defined in this document MUST NOT appear more than once in a header header section.”

**Comment by nharper** [comment]
I agree on 1 that there's no reason to exclude response pseudo-headers from this requirement. (I was worried about contradicting something regarding multiple interim responses, each of which needs a `:status` field, but each of those are in a separate field section, so a rule of no duplicates within a field section should work for pseudo-headers, both request and response.)

However, I think that a general prohibition is a good thing. If a future (negotiated) extension wants to introduce a new pseudo-header that can be duplicated, it could override an otherwise general prohibition in this draft. If a future extension doesn't do that, it gets the more sane/safe default for free.

**Comment by Lukasa** [comment]
Hmm, yeah, I think I can see a justification for that argument. Given that adding new pseudo-headers needs to be negotiated _anyway_, I don't think that's a big deal. So I'm happy to go ahead with the change for (1) and skip (2).

**Review by martinthomson — APPROVED** [review]
I had something else.  This is better.

**Comment by nharper** [comment]
@martinthomson is "this" the current text, or revised text to address Cory's comment?

**Comment by martinthomson** [comment]
This pull request.  I think that a blanket "at most once" is fine and what you have is enough.  Future extensions can deal with the consequences of that if they want >1.

**Comment by Lukasa** [comment]
@martinthomson Do you want the change to cover responses as well?

**Comment by martinthomson** [comment]
That would mean moving it up to the "HTTP Control Data" section.  That is a good idea.  I missed that this was under requests only.

**Comment by nharper** [comment]
I moved the language up to the HTTP Control Data section and slightly tweaked it.

File: draft-ietf-httpbis-http2bis.xml [hunk 1]
```diff
@@ -2829,6 +2829,9 @@ cookie: e=f
           All pseudo-header fields MUST appear in a field block before all regular field lines.
           Any request or response that contains a pseudo-header field that appears in a field
           block after a regular field line MUST be treated as <xref target="malformed">malformed</xref>.
+          The same pseudo-header field name MUST NOT appear more than once in a field block. A
```
Comment (martinthomson) [inline draft-ietf-httpbis-http2bis.xml:2832]:
```suggestion
        </t>
        <t>
          The same pseudo-header field name MUST NOT appear more than once in a field block. A
```

Let's use paragraphs.

**Review by martinthomson — APPROVED** [review]

**Review by Lukasa — APPROVED** [review]

**Pull request merged**